packages/atomic/src/components/search/result-template-components/atomic-quickview-sidebar/atomic-quickview-sidebar.tsx

```diff
@@ -86,47 +92,86 @@ const HighlightKeywordsCheckbox: FunctionalComponent<
   </Fragment>
 );
 
-const KeywordsNavigation: FunctionalComponent<
-  Pick<QuickviewSidebarProps, 'i18n'> & {
+const Keywords: FunctionalComponent<
+  Pick<
+    QuickviewSidebarProps,
+    'i18n' | 'onHighlightKeywords' | 'highlightKeywords'
+  > & {
     words: Record<string, QuickviewWordHighlight>;
   }
-> = ({words, i18n}) => {
+> = ({words, i18n, highlightKeywords, onHighlightKeywords}) => {
   return (
     <Fragment>
       {Object.values(words).map((keyword) => {
+        const wordIsEnabled =
+          highlightKeywords.highlightAll &&
```

Comment:
`highlightAll` implies to me that a value of `true` would mean that all keywords are highlighted.

In this case, I think the name `enabled` or flipping it to something like `highlightNone` would be more intuitive.